controllers/post_start.go

```diff
@@ -0,0 +1,115 @@
+package controllers
+
+/*
+Copyright 2022 The k8gb Contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Generated by GoLic, for more details see: https://github.com/AbsaOSS/golic
+*/
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/k8gb-io/k8gb/controllers/depresolver"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (r *GslbReconciler) PostStartHook(operatorConfig *depresolver.Config, mgr ctrl.Manager) error {
+	if err := inferClusterGeoTag(operatorConfig, mgr.GetAPIReader()); err != nil {
+		log.Err(err).Msgf("Can't infer the %s", depresolver.ClusterGeoTagKey)
+		return err
+	}
+	if err := createOrUpdateExternalDNSConfigMap(operatorConfig, mgr.GetClient()); err != nil {
+		log.Err(err).Msgf("Can't create/update config map for external-dns")
+		return err
+	}
```

Comment:
I cannot fully evaluate if it would not be better from the maintenance point of view to always create configMap declaratively (somewhere in yaml configurations) with default value `fmt.Sprintf("%s-%s-%s", txtOwnerPrefix, cfg.DNSZone, cfg.ClusterGeoTag)`. And only if `cfg.Route53HostedZoneID` was set, we would update the txtOwnerKey. 

@team WDYT?

Reply:
In the case when geo tag is not provided upfront, we have it only after k8gb starts (reading it from the node labels), so we can't have it in yaml in helm chart.

why not to create the cm before external-dns start and update the key from the k8gb?
..because the if the env var in Deployment is [referenced](https://github.com/k8gb-io/k8gb/pull/823/files#diff-c5b1eb253f3b95c48e2f9203764d56752b157636d69930e79e5e7e27a59567d0R39) from a ConfigMap and that cm is not there yet, the deployment will wait for the cm to be created. So it solves the race/data condition when external-dns could do its job (reconcile the dnsendpoints) with wrong ownerId + updating the CM will not make the deployment that uses it to restart iirc, we would need to restart it anyway.

I am currently looking into an option to throw away the geo tag from the `txt-owner-id` as we talked about it and use the `k8gb-{{ .Values.k8gb.dnsZone }}-$uuid` where `$uuid` is generated during the helm install (w/ the option to override it using helm values).. the problem with this approach is that helm upgrade will regenerate the uuid (as I've pointed out during the call) so it's quite cumbersome to do that... helm install will give you some random id, then we have to use it during the helm upgrade ... not very user (/dev-ops) friendly

Reply:
@jkremser , ok, thanks for the explanation